packages/react/src/components/Modal/index.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { PropsWithChildren, useEffect } from "react";
+import classNames from "classnames";
+import { createPortal } from "react-dom";
+import { Button } from ":/components/Button";
+
+export type ModalHandle = {};
+
+export enum ModalSize {
+  SMALL = "small",
+  MEDIUM = "medium",
+  LARGE = "large",
+  FULL = "full",
+}
+
+export const useModal = ({
+  isOpenDefault,
+}: { isOpenDefault?: boolean } = {}) => {
+  const [isOpen, setIsOpen] = React.useState(!!isOpenDefault);
+  const onClose = () => {
+    setIsOpen(false);
+  };
+
+  const open = () => {
+    setIsOpen(true);
+  };
+
+  const close = () => {
+    onClose();
+  };
+
+  return {
+    isOpen,
+    onClose,
+    open,
+    close,
+  };
+};
```

Comment:
I do not understand the purpose of `onClose` method.

Reply:
The `onClose` function is the name of the corresponding Modal's props. It is called when the user closes the modal, and the `close` function is just a sugar function to be used when we want to programmatically close the modal from the parent. ( It makes more sense to call `close()` instead of `onClose()` in this situation )